File: upp-service/src/main/java/rs/ac/uns/ftn/uppservice/dto/response/BookDto.java
package rs.ac.uns.ftn.uppservice.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class BookDto {

    private Long id;
    private String title;
    private String authorsName;
    private String genre;
    private Boolean isPublished;
    private String processInstanceId;
    private String handwritePath;
    private Set<BetaReaderCommentDto> betaReadersComments;

    private String jurisdiction;

    /*This field is null until second book form is submitted*/
    private String cityCountry;
}

File: upp-frontend/src/app/model/dto/BookDto.ts
import { BetaReaderCommentDto } from "./BetaReaderCommentDto";

export interface BookDto {
    id: string;
    title: string;
    authorsName: string;
    genre: string;
    isPublished: boolean;
    processInstanceId: string;
    cityCountry: string;
    handwritePath: string;
    betaReadersComments: BetaReaderCommentDto[];
    isPlagiarized: boolean;
    isRejected: boolean;

    jurisdiction: string;
}
